### PR TITLE
ci: Add obs testing for packaging

### LINKED ISSUE #621: package testing: Add basic testing for more distros
It may be possible to do a quick test in docker containers to to a kata-env to verify all the components are in place.

### PR DESCRIPTION
**Pull request by GabyCT**
This will test that is possible to install the obs packages in different
distributions.

Fixes #621

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

**Comment by GabyCT** [comment]
@jcvenegas should this test should run in a different jenkins job ?

**Comment by GabyCT** [comment]
@jcvenegas, I did the changes that we discussed

File: Makefile [hunk 1]
```diff
@@ -49,6 +49,9 @@ snap: $(YQ) $(VERSION_FILE)
 	fi
 	snapcraft -d
 
+obs-test:
+	bash tests/run_obs_testing.sh
```
Comment (jcvenegas) [inline Makefile]: Nit: $(MK_DIR)

File: run_obs_testing.sh [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+install_packages() {
+	for i in "${OS_DISTRIBUTION[@]}"; do
+		echo "Test OBS packages for ${OS_DISTRIBUTION}"
+		run_test "${i}" "${DOCKERFILE_PATH}"
+		remove_image_and_dockerfile "${i}" "${DOCKERFILE_PATH}"
+	done
+}
+
+run_test() {
+	local OS_DISTRIBUTION=${1:-}
```
Comment (jcvenegas) [inline run_obs_testing.sh]: Nit: Miss to check the second parameter

File: run_obs_testing.sh [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	generate_dockerfile "${OS_DISTRIBUTION}" "${DOCKERFILE_PATH}"
+	build_dockerfile "${OS_DISTRIBUTION}" "${DOCKERFILE_PATH}"
+}
+
+
+generate_dockerfile() {
+	local OS_DISTRIBUTION=${1:-}
```
Comment (jcvenegas) [inline run_obs_testing.sh]: Nit: Miss to check second arg

File: run_obs_testing.sh [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+build_dockerfile() {
+	local OS_DISTRIBUTION=${1:-}
+
+	pushd "${DOCKERFILE_PATH}"
+		sudo docker build \
+			--build-arg http_proxy="${http_proxy}" \
+			--build-arg https_proxy="${https_proxy}" \
+			--tag "test-${OS_DISTRIBUTION}" .
```
Comment (jcvenegas) [inline run_obs_testing.sh]: nit: I would name `obs-kata-test-${OS_DISTRIBUTION}` to help to identify the images from this test

**Comment by GabyCT** [comment]
@jcvenegas changes applied

**Pull request merged**